pkg/registry/entry.go

```diff
@@ -129,8 +132,25 @@ func (u *urlEntry) Contains(e Entry) bool {
 	return false
 }
 
-func (u *urlEntry) Match(s string) bool {
-	return u.Contains(ParseEntry(s))
+func (u *urlEntry) Match(s string) (result bool) {
+	q := ParseEntry(s)
+	query, ok := q.(URLEntry)
+	if !ok {
+		return false
+	}
+
+	result = u.Contains(query)
+	if result {
+		return
+	}
+
+	if query.URL().Scheme != "" && u.URL().Scheme == "" {
+		query.URL().Scheme = u.URL().Scheme
+	} else if u.URL().Scheme != "" && query.URL().Scheme == "" {
+		u.URL().Scheme = query.URL().Scheme
+	}
+
+	return u.Contains(query)
```

Comment:
Sure, yeah I guess `Contains` could potentially be expensive. I thought it'd make sense to try `Contains` before any special logic, in case it Just Matches™, but maybe not